plugin/color.py

```diff
@@ -0,0 +1,54 @@
+from .core.edit import parse_text_edit
+from .core.protocol import ColorPresentation
+from .core.protocol import ColorPresentationParams
+from .core.protocol import Request
+from .core.registry import LspTextCommand
+from .core.typing import List
+from .core.views import range_to_region
+import sublime
+
+
+class LspColorPresentationCommand(LspTextCommand):
+
+    capability = 'colorProvider'
+
+    def run(self, edit: sublime.Edit, params: ColorPresentationParams) -> None:
+        session = self.best_session(self.capability)
+        if session:
+            self._version = self.view.change_count()
+            self._range = params['range']
+            session.send_request_async(Request.colorPresentation(params, self.view), self._handle_response_async)
+
+    def want_event(self) -> bool:
+        return False
+
+    def _handle_response_async(self, response: List[ColorPresentation]) -> None:
+        if not response:
+            return
+        window = self.view.window()
+        if not window:
+            return
+        if self._version != self.view.change_count():
+            return
+        old_text = self.view.substr(range_to_region(self._range, self.view))
+        self._filtered_response = []  # type: List[ColorPresentation]
+        for item in response:
+            # Filter out items that would apply no change
+            text_edit = item.get('textEdit')
+            if text_edit:
+                if text_edit['range'] == self._range and text_edit['newText'] == old_text:
```

Comment:
Hm, what makes you think that this would be the case?

According to https://stackoverflow.com/a/40921229 this is the correct way to compare equality of keys and values in Python dicts.

And I can confirm that it works with LSP-css, i.e. the server does use `textEdit`s and the unnecessary item is correctly filtered out.

Reply:
Well, Python is smarter than I thought. Never mind.